### PR TITLE
remove dotenv dependency

### LINKED ISSUE #1811: RUSTSEC-2021-0141: dotenv is Unmaintained
> dotenv is Unmaintained

| Details             |                                                |
| ------------------- | ---------------------------------------------- |
| Status              | unmaintained                |
| Package             | `dotenv`                      |
| Version             | `0.15.0`                   |
| URL                 | [https://github.com/dotenv-rs/dotenv/issues/74](https://github.com/dotenv-rs/dotenv/issues/74) |
| Date                | 2021-12-24                         |

dotenv by description is meant to be used in development or testing only.

Using this in production may or may not be advisable.

## Alternatives

The below may or may not be feasible alternative(s):
- [dotenvy](https://crates.io/crates/dotenvy)

See [advisory page](https://rustsec.org/advisories/RUSTSEC-2021-0141.html) for additional details.

### PR DESCRIPTION
**Pull request by syphar**
resolves #1811 

We need to add the config to the tests explicitly, which personally I like more than how it was done before.

### CODE REVIEW

**Comment by syphar** [comment]
r? @Nemo157 

also @jyn514 (?) to know if this is really not used in prod?

**Comment by Nemo157** [comment]
In production the systemctl unit is configured with:

```ini
EnvironmentFile=/home/cratesfyi/.docs-rs-env
WorkingDirectory=/home/ubuntu/docs.rs
```

The `.env` loading is relative to working directory (and walks up the directory tree), so checking for that:

```console
> ls /home/ubuntu/docs.rs/.env /home/ubuntu/.env /home/.env /.env
ls: cannot access '/home/ubuntu/docs.rs/.env': No such file or directory
ls: cannot access '/home/ubuntu/.env': No such file or directory
ls: cannot access '/home/.env': No such file or directory
ls: cannot access '/.env': No such file or directory
```

**Comment by syphar** [comment]
@Nemo157 thanks for checking! that sounds like we're safe removing this :)

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -8,6 +8,16 @@ on:
...
+  DOCSRS_PREFIX: ignored/cratesfyi-prefix
+  DOCSRS_DATABASE_URL: postgresql://cratesfyi:password@localhost:15432
+  DOCSRS_LOG: docs_rs=debug,rustwide=info
+  AWS_ACCESS_KEY_ID: cratesfyi
+  AWS_SECRET_ACCESS_KEY: secret_key
+  S3_ENDPOINT: http://localhost:9000
+  DOCSRS_INCLUDE_DEFAULT_TARGETS: false
+  DOCSRS_DOCKER_IMAGE: ghcr.io/rust-lang/crates-build-env/linux-micro
+  SENTRY_ENVIRONMENT: dev
+  DOCSRS_METADATA_HOST_TARGET: aarch64-unknown-linux-gnu
```
Comment (Nemo157) [inline .github/workflows/ci.yml:20]: This shouldn't be necessary, it's set by the `build.rs`.

Comment (syphar) [reply .github/workflows/ci.yml]: @Nemo157 I removed the line

**Review by Nemo157 — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -64,6 +64,10 @@ mkdir -p ignored/cratesfyi-prefix/crates.io-index
 cargo build
 # Start the external services
 docker-compose up -d db s3
+# anything that doesn't run via docker-compose needs the settings defined in 
+# .env. Either via `. .env` as below, or via any dotenv shell integration
+(dotenv, direnv, ...).
```
Comment (jyn514) [inline README.md:69]: this isn't valid syntax, I would rather just have the comment.
```suggestion
```

**Review by jyn514 — APPROVED** [review]
this seems fine, I guess. I'm probably going to be a bit annoyed by it next time I run docs.rs locally, but I also do that so infrequently I don't think my opinion should matter very much.

File: README.md (hunk 2 above)
Comment (syphar) [reply README.md]: valid point, I removed it.

**Pull request merged**